buildSrc/subprojects/parser/src/main/java/org/apache/calcite/buildtools/parser/DialectGenerate.java

```diff
@@ -102,26 +102,24 @@ public void unparseReservedKeywords(ExtractedData extractedData) {
     if (extractedData.keywords.isEmpty()) {
       return;
     }
-    boolean first = true;
     StringBuilder stringBuilder = new StringBuilder();
     stringBuilder.append("<DEFAULT, DQID, BTID> TOKEN :\n{\n");
+    String tokenTemplate = "<%s : \"%s\">";
+    List<String> tokens = new LinkedList<String>();
     for (Map.Entry<Keyword, String> entry : extractedData.keywords.entrySet()) {
+      StringBuilder tokenBuilder = new StringBuilder();
       Keyword keyword = entry.getKey();
-      String tokenTemplate = "| <%s : \"%s\">";
-      if (first) {
-        tokenTemplate = "<%s : \"%s\">";
-        first = false;
-      }
-      String token = String.format(tokenTemplate, keyword.keyword,
-          entry.getValue());
-      stringBuilder.append(token);
+      tokenBuilder.append(String.format(tokenTemplate, keyword.keyword,
+          entry.getValue()));
       if (keyword.filePath == null) {
-        stringBuilder.append(" // No file specified.");
+        tokenBuilder.append(" // No file specified.");
       } else {
-        stringBuilder.append(" // From: ").append(keyword.filePath);
+        tokenBuilder.append(" // From: ").append(keyword.filePath);
       }
-      stringBuilder.append("\n");
+      tokenBuilder.append("\n");
+      tokens.add(tokenBuilder.toString());
     }
+    stringBuilder.append(String.join("| ", tokens));
```

Comment:
Does the current StringJoin logic work with the comments above? I may just be reading this incorrectly, but it seems like it would put the "|" character after the comment?

Reply:
Maybe we can include a test in this PR to illustrate that this works?

Reply:
It does put it after but there is a new line after each token so it works out.

Reply:
Ah true.

Reply:
There is already a test here: https://github.com/googleinterns/calcite/blob/619cd86e3738c1a74de61b176947b2655544692f/buildSrc/subprojects/parser/src/test/java/org/apache/calcite/test/DialectGenerateTest.java#L367

Reply:
Thanks, I missed that too :)